tests/dotnet/UnitTests/AssetsTest.cs

```diff
@@ -0,0 +1,221 @@
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
+		}
+
+		void ConfigureAssets ()
+		{
+			// We either want the assets added before the build, or we will be adding them after the build
+			if (isStartingWithAssets)
+				CopyAssets ();
+
+			Clean (projectPath);
+
+			DotNet.AssertBuild (projectPath, GetDefaultProperties (runtimeIdentifiers));
+			if (!isStartingWithAssets) {
+				CopyAssets ();
+				// Building the project twice without cleaning in between fails: https://github.com/xamarin/maccore/issues/2530
+				Clean (projectPath);
+				DotNet.AssertBuild (projectPath, GetDefaultProperties (runtimeIdentifiers));
+			}
+		}
+
+		void DeleteAssets ()
+		{
+			var xcassetsDir = Path.Combine (projectPath, "../Assets.xcassets");
+			File.Delete (xcassetsDir);
+		}
+
+		void CopyAssets ()
+		{
+			var testingAssetsDir = new DirectoryInfo (Path.Combine (projectPath, "../../TestingAssets"));
+			var xcassetsDir = new DirectoryInfo (Path.Combine (projectPath, "../Assets.xcassets"));
+
+			Assert.That (testingAssetsDir, Does.Exist, $"Could not find testingAssetsDir: {testingAssetsDir}");
+			MakeSymlinks (testingAssetsDir.FullName, xcassetsDir.FullName);
+			Assert.That (xcassetsDir, Does.Exist, $"Could not find xcassetsDir: {xcassetsDir}");
+		}
+
+		void MakeSymlinks (string sourceDir, string destDir)
+		{
+			var output = new StringBuilder ();
+			var executable = "ln";
+			var arguments = new string [] { "-s", sourceDir, destDir };
+			var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (60)).Result;
+			Assert.AreEqual (0, rv.ExitCode, $"Creating Symlink Error: {rv.StandardError}. Unexpected ExitCode");
+		}
+
+		string GetFullSdkVersion () => sdkVersion switch {
+			"iphonesimulator" => sdkVersion + Configuration.sdk_version,
+			"iphoneos" => sdkVersion + Configuration.sdk_version,
+			"appletvsimulator" => sdkVersion + Configuration.tvos_sdk_version,
+			"macosx" => sdkVersion + Configuration.macos_sdk_version,
+			_ => throw new ArgumentOutOfRangeException (nameof (sdkVersion), $"Not expected sdkVersion: {sdkVersion}"),
+		};
+
+		JsonDocument ProcessAssets (string assetsPath, string sdkVersion)
+		{
+			var output = new StringBuilder ();
+			var executable = "xcrun";
+			var arguments = new string [] { "--sdk", sdkVersion, "assetutil", "--info", assetsPath };
+			var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (120)).Result;
+			Assert.AreEqual (0, rv.ExitCode, $"Processing Assets Error: {rv.StandardError}. Unexpected ExitCode");
+			var s = output.ToString ();
+
+			// This Execution call produces an output with an objc warning. We just want the json below it.
+			if (s.StartsWith ("objc"))
+				output.Remove (0, s.IndexOf (Environment.NewLine) + 1);
```

Comment:
You can just create a new string like this:

```suggestion
				s = s.Substring (s.IndexOf (Environment.NewLine) + 1);
```